syntax = "proto3";

service ProductImageAnalyzer {
  rpc GenerateFromImage (ImageRequest) returns (ImageResponse);
  rpc GenerateFromImageUrl (ImageUrlRequest) returns (ImageResponse);
}

message ImageRequest {
  bytes image = 1;
  string filename = 2;
  string content_type = 3;
}

message ImageUrlRequest {
  string image_url = 1;
}

message ImageResponse {
  string title = 1;
  string description = 2;
}